tests/index.test.ts

```diff
@@ -134,50 +181,49 @@ describe('ParaSwap SDK', () => {
 
     expect(Array.isArray(others)).toBe(true);
 
-    expect(typeof others![0].exchange).toBe('string');
+    const firstRoute = others?.[0];
 
-    expect(typeof others![0].unit).toBe('string');
-    expect(new BigNumber(others![0].unit as string).isNaN()).toBe(false);
+    assert(firstRoute, 'at least one route must exist');
+
+    expect(typeof firstRoute.exchange).toBe('string');
+
+    expect(typeof firstRoute.unit).toBe('string');
+    expect(firstRoute.unit && new BigNumber(firstRoute.unit).isNaN()).toBe(
+      false
+    );
   });
 
   test('Get_Spender', async () => {
-    const spender = await paraSwap.getTokenTransferProxy();
-    expect(provider.utils.isAddress(spender as string));
+    const spender = await paraSwap.getSpender();
+    expect(web3provider.utils.isAddress(spender));
   });
 
   test('Get_Allowance', async () => {
-    const allowance = await paraSwap.getAllowance(senderAddress, DAI);
+    const allowance = await paraSwap.getAllowance(
+      DUMMY_ADDRESS_FOR_TESTING_ALLOWANCES,
+      DAI
+    );
 
-    if (!allowance || (allowance as APIError).message) {
-      return;
-    }
+    assert(isAllowance(allowance), 'hardcoded dummy address should be found');
 
-    expect(new BigNumber((allowance as Allowance).allowance).isNaN()).toBe(
-      false
-    );
+    expect(allowance.allowance).toEqual('123000000000000000');
```

Comment:
Whose address is dummy address? Can you guarantee that its allowance won't change from this fixed number?
If you can't, maybe we should rather `approve, then expect(allowance)`, that would be more reliable

Reply:
This dummy address I've specifically created and adjusted for the sake of these tests.
I can throw out the PK in order to guarantee it'd never change.

The issue with `approve` and then `expect(allowance)` is that the allowance is taken from the API, and API would return info based on mainnet, not the ganache fork we're dealing with in the tests